### ISSUE TITLE
Create a consistent way of referring to the MaskRCNN implementation

### ISSUE THREAD

**Issue by havanagrawal** [havanagrawal/clomask#17]
Since the matterport implementation is not available via a package manager,  we want a way to consistently refer to the library from code.

- One solution is to simply clone the repository as a subdirectory in this one, the [license](https://github.com/matterport/Mask_RCNN/blob/master/LICENSE) allows us to do this. The pro is that this is the simplest solution, the con is that it will bloat our own repo. We could strip out the parts that we don't need to alleviate that issue.
- Another solution is to ensure that everyone has an environment variable that points to their local directory for MaskRCNN. For example:
`MASK_CNN_DIR='/Users/havan/Dropbox/CP/Git/MaskRCNN'`
We would then add all necessary paths from code, preferably from a central module/lib:
```python
mask_rcnn_path = os.environ.get('MASK_CNN_DIR')
sys.path.append(os.path.join(mask_rcnn_path, 'mrcnn'))
sys.path.append(os.path.join(mask_rcnn_path, 'samples', 'coco'))

import utils
import coco
...
```

A subsequent task would be to move all model related code to a central library that we can simply import from. Computing `mAP` (mean average precision) is one such piece of code, that is currently duplicated across notebooks.
 
Please let me know what your thoughts are on this. Mentioning @pshivraj since we discussed this offline.

**Comment by lmtoan** [comment]
Thanks Havan, @pshivraj 

My thought is that we should clone the [mrcnn](https://github.com/matterport/Mask_RCNN/tree/master/mrcnn) module and incorporate it into our `clomask` repo. That way, we can easily customize any training/inference class if needed. The mrcnn module only has 5 scripts so it shouldn't bloat our repo that much. We can put all our research code in a `research` folder, similar to one of their `samples/xxx` folder.  

When I interned at Clobotics, they copied the giant Caffe repo and developed their own stuff in parallel folders. At the end of the Capstone, I think they would prefer us passing on a custom `mrcnn` module with detailed requirements/installation instructions, without referencing back and forth to `matterport` repo.

Let's discuss more.

**Comment by havanagrawal** [comment]
Fixed by #32, #36
